### PR TITLE
"at" module add chdir option

### LINKED ISSUE #13: [RFE] Add `chdir` parm to the `at` module
##### SUMMARY

This RFE is to have a `chdir` parameter, similar to what's currently available for the `shell` module, added to the `at` module.

When using `at` with the `ansible-pull` and the `--purge` option the job will fail since the execution directory is being removed. 

As a workaround, the `at` job can be scheduled using the shell module. 

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

at.py
##### ADDITIONAL INFORMATION

cmd:
```
ansible-pull -U https://github.com/benjdudas/ansible_pull.git -i hosts --purge
```

local.yml
```yaml
---
- hosts: local
  connection: local
  gather_facts: false

  tasks:
    - name: run logger in 2 minutes
      at:
        command: /bin/logger 'AT task ran from Ansible'
        count: 2
        units: minutes
        unique: yes
      environment:
        PWD: /root

    - name: copy job file
      copy:
        src: ./myjob.txt
        dest: /root/myjob.txt

    - name: testing with shell mod, runs date in 2 min
      shell:
        chdir: /root
        cmd: at now + 2 min -f myjob.txt

    - name: cleanup job file
      file:
        path: /root/myjob.txt
        state: absent

...
```
mail logs:
```
From root@rhel-awx.localdomain  Tue Apr 14 11:59:00 2020
Return-Path: <root@rhel-awx.localdomain>
X-Original-To: vagrant
Delivered-To: vagrant@rhel-awx.localdomain
Received: by rhel-awx.localdomain (Postfix, from userid 0)
        id 9F4431067836; Tue, 14 Apr 2020 11:59:00 -0400 (EDT)
Subject: Output from your job       36
To: vagrant@rhel-awx.localdomain
Message-Id: <20200414155900.9F4431067836@rhel-awx.localdomain>
Date: Tue, 14 Apr 2020 11:59:00 -0400 (EDT)
From: root@rhel-awx.localdomain (root)

sh: line 31: cd: /root/.ansible/pull/rhel-awx: No such file or directory
Execution directory inaccessible

From root@rhel-awx.localdomain  Tue Apr 14 12:00:00 2020
Return-Path: <root@rhel-awx.localdomain>
X-Original-To: vagrant
Delivered-To: vagrant@rhel-awx.localdomain
Received: by rhel-awx.localdomain (Postfix, from userid 0)
        id CCF951067836; Tue, 14 Apr 2020 12:00:00 -0400 (EDT)
Subject: Output from your job       38
To: vagrant@rhel-awx.localdomain
Message-Id: <20200414160000.CCF951067836@rhel-awx.localdomain>
Date: Tue, 14 Apr 2020 12:00:00 -0400 (EDT)
From: root@rhel-awx.localdomain (root)

Tue Apr 14 12:00:00 EDT 2020
```

The first mail was from the `at` task (run logger in 2 minutes)  showing "Execution directory inaccessible". The second mail log is from the shell task (testing with shell mod, runs date in 2 min)  running /bin/date with `at`.

### PR DESCRIPTION
**Pull request by bmillemathias**
##### SUMMARY
Add parameter chdir to module `at`
Fixes #13 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
at

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/at.py [hunk 1]
```diff
@@ -21,21 +21,15 @@
     description:
      - A command to be executed in the future.
     type: str
-  script_file:
```
Comment (gundalow) [inline plugins/modules/at.py:24]: I don't see the new `chdir` option here.
May I suggest that you don't change the order of the options here as it causes a noisy git-diff

Comment (bmillemathias) [reply plugins/modules/at.py:24]: still in WIP, will work on that this week-end.

Comment (gundalow) [reply plugins/modules/at.py:24]: Thanks, was just taking a quick skim over the open PRs to see if I could help.

Comment (bmillemathias) [reply plugins/modules/at.py:24]: >  May I suggest that you don't change the order of the options here as it causes a noisy git-diff

Ok will change that

File: changelogs/fragments/102_module_at_add_chdir_option.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - module at - add option chdir to permit to launch the at command from a specific directory
```
Comment (aminvakil) [inline changelogs/fragments/102_module_at_add_chdir_option.yml:2]:
```suggestion
  - module at - add option ``chdir`` to permit to launch the at command from a specific directory
```

**Comment by quidame** [comment]
@bmillemathias is this work still in progress ?

**Comment by bmillemathias** [comment]
> @bmillemathias is this work still in progress ?

Looking at the code it seems to be ready, do you want to review it ?

File: plugins/modules/at.py [hunk 3]
```diff
@@ -188,7 +196,8 @@ def main():
     result['count'] = count
     result['units'] = units
 
-    add_job(module, result, at_cmd, count, units, command, script_file)
+    add_job(module, result, at_cmd, count, units, command, script_file,
+            chdir=chdir)
```
Comment (quidame) [inline plugins/modules/at.py:200]: why a line return for so few ?

File: tests/integration/targets/at/tasks/main.yml [hunk 4]
```diff
@@ -54,9 +54,55 @@
...
+
+- name: add a at useless command
+  at:
+    command: /bin/logger 'AT task ran from Ansible'
+    count: 1
+    units: minutes
+    unique: true
+    state: absent
+  register: at_removal
+  ignore_errors: true
```
Comment (quidame) [inline tests/integration/targets/at/tasks/main.yml:86]: is the task supposed to fail ?

File: tests/integration/targets/at/tasks/main.yml [hunk 5]
```diff
@@ -54,9 +54,55 @@
...
+- name: create an at command with chdir with invalid value
+  at:
+    command: /bin/logger 'AT task ran from Ansible with chdir'
+    count: 1
+    units: minutes
+    chdir: /invalid
+  register: at_chdir_invalid
+  ignore_errors: true
+
+- debug: var=at_chdir_invalid
```
Comment (quidame) [inline tests/integration/targets/at/tasks/main.yml:108]: There are 2 `ignore_errors: true` but nothing to manage/handle the how and why the failure, and even to ensure a failure happened when expected. Typically, such tasks should end with an `assert` task to check previously registered results, as above.

**Review by quidame — COMMENTED** [review]
@bmillemathias thanks for implementing this ! I can review but not merge, so other people would take a look, I hope soon.

File: plugins/modules/at.py [hunk 6]
```diff
@@ -17,6 +17,12 @@
  - All jobs are executed in the 'a' queue.
 version_added: "1.0.0"
 options:
+  chdir:
+    description:
+    - An optional location from where to run the command at. Useful for intance
+      when running a playbook using ansible-pull with C(purge) option.
+    type: path
+    version_added: 1.1.2
```
Comment (quidame) [inline plugins/modules/at.py:25]:
```suggestion
    version_added: 1.3.0
```

File: plugins/modules/at.py [hunk 7]
```diff
@@ -17,6 +17,12 @@
  - All jobs are executed in the 'a' queue.
 version_added: "1.0.0"
 options:
+  chdir:
+    description:
+    - An optional location from where to run the command C(at). Useful for intance
```
Comment (Akasurde) [inline plugins/modules/at.py:22]:
```suggestion
    - An optional location from where to run the command C(at).
    - Useful for instance when running a playbook using ansible-pull with C(purge) option.
```

**Comment by Akasurde** [comment]
@bmillemathias Are you still working on this? Let me know otherwise. Thanks,

**Comment by bmillemathias** [comment]
> @bmillemathias Are you still working on this? Let me know otherwise. 

Hi  @Akasurde you can take it,

thanks for the support

**Comment by Akasurde** [comment]
> > @bmillemathias Are you still working on this? Let me know otherwise.
> 
> Hi @Akasurde you can take it,
> 
> thanks for the support

Thanks, @bmillemathias for the contribution.

**Pull request closed without merging**